templates/.github/workflows/go-check.yml

```diff
@@ -48,4 +56,17 @@ jobs:
           run: |
             set -o pipefail
             staticcheck ./... | sed -e 's@\(.*\)\.go@./\1.go@g'
+      - name: go generate
+        uses: protocol/multiple-go-modules@v1.2
+        if: env.RUNGOGENERATE == 'true'
+        with:
+          run: |
+            git clean -fd # make sure there aren't untracked files / directories
+            go generate ./...
+            gitstatus=$(git status --porcelain)
```

Comment:
Please pay attention to these lines when reviewing. Is this sane?

Reply:
Ah, I see. We can't use `git diff --exit-code` because of untracked files?

This seems sane, but I'd also consider `git add . && git diff --exit-code`.

Reply:
From stackoverflow, I found `git ls-files --exclude-standard --others`, which I think I'd prefer slightly as it has a narrower purpose. If that doesn't work, I'd prefer Steven's suggestion with add+diff over just status, since it seems more intuitive.

Reply:
Wait, I'm confused now. Exactly what files are you checking for, here? Unstaged changes, *excluding* untracked and ignored files? Then my suggestion above might be wrong. I got it from https://stackoverflow.com/a/2659808/4931501, where there are other ideas depending on what you want.

Would also be good to document this inline, because I think I already misunderstood the code once :)

Reply:
There are two things that should fail the build:
1. When `go generate` makes a change to a tracked file. Obviously, the PR should have updated that file.
2. When `go generate` adds a new file. The PR probably forgot to commit the generated file.

Ignored files shouldn't be considered. They're ignored for a reason.

If I understand correctly, `git ls-files --exclude-standard --others` does 2., but not 1.
`git add . && git diff --exit-code` doesn't work, since as soon as a file is `git add`ed, it doesn't show up in `git diff` any more.

Reply:
As far as I can tell, `git add . && git diff-index HEAD` would do 1. and 2. I guess that's nicer than `git status --porcelain`.

Reply:
Even better, it allows using the `--exit-code` flag, so we don't have to do string comparisons.

Reply:
FYI there is also diff --cached, aka --staged.